Incorrect line number/column in F823 error

For code:
```py
TOMATO = "black cherry"


def update_tomato():
    print(TOMATO)
    TOMATO = "cherry tomato"


update_tomato()
```

I get output:
```
❯ flake8 repro.py 
repro.py:5:11: F823 local variable 'TOMATO' defined in enclosing scope on line 1 referenced before assignment
repro.py:6:5: F841 local variable 'TOMATO' is assigned to but never used

❯ ruff repro.py   
repro.py:6:5: F823 Local variable `TOMATO` referenced before assignment
repro.py:6:5: F841 Local variable `TOMATO` is assigned to but never used

Found 2 error(s).
```
flake8 shows the correct position while ruff doesn't.

--- a/resources/test/fixtures/F821.py
+++ b/resources/test/fixtures/F821.py
@@ -77,3 +77,8 @@ class Ticket:
 
     def set_status(self, status: Status):
         self.status = status
+
+
+def update_tomato():
+    print(TOMATO)
+    TOMATO = "cherry tomato"

--- a/src/ast/types.rs
+++ b/src/ast/types.rs
@@ -53,5 +53,7 @@ pub enum BindingKind {
 pub struct Binding {
     pub kind: BindingKind,
     pub location: Location,
-    pub used: Option<usize>,
+    /// Tuple of (scope index, location) indicating the scope and location at which the binding was
+    /// last used.
+    pub used: Option<(usize, Location)>,
 }

--- a/src/check_ast.rs
+++ b/src/check_ast.rs
@@ -184,7 +184,7 @@ where
                                 name.to_string(),
                                 Binding {
                                     kind: BindingKind::Assignment,
-                                    used: Some(global_scope_id),
+                                    used: Some((global_scope_id, stmt.location)),
                                     location: stmt.location,
                                 },
                             );
@@ -416,13 +416,14 @@ where
                             name,
                             Binding {
                                 kind: BindingKind::FutureImportation,
-                                used: Some(
+                                used: Some((
                                     self.scopes[*(self
                                         .scope_stack
                                         .last()
                                         .expect("No current scope found."))]
                                     .id,
-                                ),
+                                    stmt.location,
+                                )),
                                 location: stmt.location,
                             },
                         );
@@ -1108,7 +1109,7 @@ impl<'a> Checker<'a> {
                     }
                 }
                 if let Some(binding) = scope.values.get_mut(id) {
-                    binding.used = Some(scope_id);
+                    binding.used = Some((scope_id, expr.location));
                     return;
                 }
 
@@ -1136,17 +1137,14 @@ impl<'a> Checker<'a> {
             {
                 for scope in self.scopes.iter().rev().skip(1) {
                     if matches!(scope.kind, ScopeKind::Function | ScopeKind::Module) {
-                        let used = scope
-                            .values
-                            .get(id)
-                            .map(|binding| binding.used)
-                            .unwrap_or_default();
-                        if let Some(scope_id) = used {
-                            if scope_id == current.id {
-                                self.checks.push(Check::new(
-                                    CheckKind::UndefinedLocal(id.clone()),
-                                    expr.location,
-                                ));
+                        if let Some(binding) = scope.values.get(id) {
+                            if let Some((scope_id, location)) = binding.used {
+                                if scope_id == current.id {
+                                    self.checks.push(Check::new(
+                                        CheckKind::UndefinedLocal(id.clone()),
+                                        location,
+                                    ));
+                                }
                             }
                         }
                     }

